engine/src/test/java/io/camunda/zeebe/engine/processing/incident/JobUnhandledErrorEventIncidentTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.incident;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.protocol.record.value.ErrorType;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+/** Tests for incidents created due to UNHANDLED_ERROR_EVENT associated to jobs */
+public final class JobUnhandledErrorEventIncidentTest {
```

Comment:
(optional) We could merge this test with the related tests in `io.camunda.zeebe.engine.processing.incident.ErrorEventIncidentTest`.

Reply:
Interesting, I did not consider this to be about the same topic. But it is. I'll move it there.

Side note, just now discovered that one of the tests in ErrorEventIncidentTest takes 10 seconds. I might have a look at that as well

Reply:
Sounds good :+1:

Reply:
@saig0 I've cleaned-up and sped-up that test class a bit, and then moved the new test there. Please have another look. Only changed commits are the first 3 (i.e. with type `test`).